ansys/ansys-templates
# Implement project commands, improve app responsiveness, prevent ConflictError raised while changing UI values quickly

## What type of PR is this? (check all applicable)


## Description
The purpose of this PR is to implement new feature of disabling the project command buttons while command execution method is in progress. This PR includes:
- Refactoring of command execution methods to a single transaction method to handle all action requests on the oSL session.
- Created a ButtonGroup and ButtonWithTooltip class to simplify project command group creation for the UI
- Removed the ActorCommandAIO class and refactored the ProjectCommandsAIO class to handle all button interactions for the root and actor commands.
- Updated version of glow to 0.4.dev4 --> significant improvement in the App's responsiveness
- Updated version of portal to 0.4.3
- Handles ConflictError thrown when changing UI values quickly

## Related Tickets

- Closes https://github.com/Solution-Applications/solution-enablement-product-management/issues/581
- Closes https://github.com/Solution-Applications/solution-enablement-product-management/issues/548
- Fixes problem seen in: https://github.com/Solution-Applications/solution-enablement-product-management/issues/546

##  Screenshots, Recordings


## QA Instructions,

1. Checkout branch associated to this PR
2. In a .venv run: flit install to gather latest changes made to the osl-solution template
3. Create a new osl-solution using overview_project.owa: ansys-templates new solution -f owa (enter all required inputs)
4. In your terminal, set your current directory to the directory of the new solution.
5.  Setup the python virtual environment by running: python setup_environment.py -d run
6.  Run the solution and create a new project
7. Start the analysis
8. Navigate to the root component of the monitoring pages. Click on the project commands and check the all buttons are disabled while the method is running. Check the an alert is displayed once method execution is complete.  
9. Repeat step 8 using the actor commands.
